From 605d8fed90e903db4e677d43c6bd49038acdc4db Mon Sep 17 00:00:00 2001
From: Brent Murphy <bmurphy@bcmcs.net>
Date: Thu, 30 Apr 2020 20:10:42 +1000
Subject: [PATCH] [macos] allow portable mode to run

Note: portable mode should only be used for testing/dev on macos. During updates, app bundles
will replace the entire existing bundle, and any portable data will be lost.
---
 xbmc/settings/SettingsComponent.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/settings/SettingsComponent.cpp b/xbmc/settings/SettingsComponent.cpp
index e5d5b17d75b3..5d1a4d186c2a 100644
--- a/xbmc/settings/SettingsComponent.cpp
+++ b/xbmc/settings/SettingsComponent.cpp
@@ -296,7 +296,6 @@ bool CSettingsComponent::InitDirectoriesOSX(bool bPlatformDirectories)
   std::string frameworksPath = CUtil::GetFrameworksPath();
   CSpecialProtocol::SetXBMCFrameworksPath(frameworksPath);
 
-  // OSX always runs with bPlatformDirectories == true
   if (bPlatformDirectories)
   {
     // map our special drives
@@ -347,6 +346,7 @@ bool CSettingsComponent::InitDirectoriesOSX(bool bPlatformDirectories)
     std::string strTempPath = URIUtils::AddFileToFolder(appPath, "portable_data/temp");
     CSpecialProtocol::SetTempPath(strTempPath);
     CSpecialProtocol::SetLogPath(strTempPath);
+    CreateUserDirs();
   }
   CSpecialProtocol::SetXBMCBinAddonPath(appPath + "/addons");
   return true;
